### PR TITLE
Rebase release-4.12 to master

### PR DESCRIPTION
**Pull request by praveenkumar**
This pr is rebase of release-4.12 branch on master.

### CODE REVIEW

File: snc.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ if test -n "${OPENSHIFT_VERSION-}"; then
     OPENSHIFT_RELEASE_VERSION=${OPENSHIFT_VERSION}
     echo "Using release ${OPENSHIFT_RELEASE_VERSION} from OPENSHIFT_VERSION"
 else
-    OPENSHIFT_RELEASE_VERSION="$(curl -L "${MIRROR}"/candidate-4.11/release.txt | sed -n 's/^ *Version: *//p')"
+    OPENSHIFT_RELEASE_VERSION="$(curl -L "${MIRROR}"/candidate-4.12/release.txt | sed -n 's/^ *Version: *//p')"
```
Comment (cfergeau) [inline snc.sh:47]: build-patched-kao-kcmo-images.sh uses latest-4.12, not sure if this is problematic?

**Review by cfergeau — COMMENTED** [review]
A few minor comments, but looks good!

File: build-patched-kao-kcmo-images.sh [hunk 2]
```diff
@@ -88,14 +90,15 @@ function patch_and_push_image() {
...
-    git checkout --track origin/rhaos-4.11-rhel-8
-    git merge --no-ff -m "Merge commit ${vcs_ref} into rhaos-4.11-rhel-8" -m "MaxFileSize: 104857600" ${vcs_ref}
+    # Just fetch the upstream/rhaos-${OCP_VERSION}-rhel-8 instead of all the branches and tags from upstream
+    git fetch upstream rhaos-${OCP_VERSION}-rhel-8 --no-tags
+    git checkout --track origin/private-rhaos-${OCP_VERSION}-rhel-8
+    git merge --no-edit ${vcs_ref}
     git push origin HEAD
-    rhpkg container-build  --target crc-1-rhel-8-candidate
+    rhpkg container-build --scratch --target crc-1-rhel-8-candidate &> rhpkg.out
+    brew_image=$(grep -o 'registry-proxy.engineering.redhat.com.*' rhpkg.out)
```
[the remark this answers is not in the document]
Comment (praveenkumar) [reply build-patched-kao-kcmo-images.sh:99]: Yes we need to fix that and not having `scratch` any more for `crc-1-rhel-8-candidate` target.

File: snc.sh (hunk 1 above)
Comment (praveenkumar) [reply snc.sh:47]: as of now there is no `latest-4.12` until the release so we are going to keep `candidate` till and switch to `latest` once available.

File: build-patched-kao-kcmo-images.sh [hunk 3]
```diff
@@ -93,7 +93,7 @@ function patch_and_push_image() {
     # Just fetch the upstream/rhaos-${OCP_VERSION}-rhel-8 instead of all the branches and tags from upstream
     git fetch upstream rhaos-${OCP_VERSION}-rhel-8 --no-tags
     git checkout --track origin/private-rhaos-${OCP_VERSION}-rhel-8
-    git merge --no-edit ${vcs_ref}
+    git merge --no-ff -m "Merge commit ${vcs_ref} into rhaos-${OCP_VERSION}-rhel-8" -m "MaxFileSize: 104857600" ${vcs_ref}
```
[the remark this answers is not in the document]
Comment (praveenkumar) [reply build-patched-kao-kcmo-images.sh:96]: fixed, that commit need to be removed.

File: snc.sh (hunk 1 above)
Comment (cfergeau) [reply snc.sh:47]: Yes, and my point was that it's wrong for `build-patched-kao-kcmo-images.sh` to refer to `latest-${OCP_VERSION}` and that this should be changed to `candidate-${OCP_VERSION}`

**Pull request merged**